certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/action/model/CertificateActionFactory.java

```diff
@@ -80,6 +80,17 @@ public static CertificateAction create(CertificateActionSpecification actionSpec
               )
           )
           .build();
+      case REVOKE -> CertificateActionRevoke.builder()
+          .certificateActionSpecification(actionSpecification)
+          .actionRules(
+              List.of(
+                  new ActionRuleWithinCareUnit(),
+                  new ActionRuleRole(List.of(Role.DOCTOR, Role.PRIVATE_DOCTOR, Role.NURSE)),
+                  new ActionRuleStatus(List.of(Status.SIGNED)),
+                  new ActionRuleProtectedPerson()
```

Comment:
This is maybe a bit redundant. Anyone that can revoke also has right to work with protected person. Just thinking that we just add necessary rules.